src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterServiceMetrics.java

```diff
@@ -58,12 +63,37 @@ public void collectMetrics(long startTime) {
                 return;
             }
 
+            /*
+            pendingTasks API returns object of PendingClusterTask which contains insertOrder, priority, source, timeInQueue.
+                Example :
+                     insertOrder: 101,
+                     priority: "URGENT",
+                     source: "create-index [foo_9], cause [api]",
+                     timeIn_queue: "86ms"
+             */
+
+            List<PendingClusterTask> pendingTasks = ESResources.INSTANCE.getClusterService().getMasterService()
+                    .pendingTasks();
+            HashMap<String,Integer> pendingTaskCountPerTaskType = new HashMap<>();
+
+            pendingTasks.stream().forEach( pendingTask -> {
+                String pendingTaskType = pendingTask.getSource().toString().split(" ",2)[0];
+                if (pendingTaskCountPerTaskType.containsKey(pendingTaskType)) {
+                    pendingTaskCountPerTaskType.put(pendingTaskType, pendingTaskCountPerTaskType.get(pendingTaskType) + 1);
+                }
+                else{
+                    pendingTaskCountPerTaskType.put(pendingTaskType,1);
+                }
+            });
+
             value.setLength(0);
-            value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds())
-                    .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
-            value.append(new MasterPendingStatus(
-                    ESResources.INSTANCE.getClusterService().getMasterService()
-                            .numberOfPendingTasks()).serialize());
+            value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds());
+            for (String pendingTaskType : pendingTaskCountPerTaskType.keySet()){
+                value.append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+                value.append(new MasterPendingStatus(
+                        pendingTaskType,pendingTaskCountPerTaskType.get(pendingTaskType)).serialize());
+            }
```

Comment:
How about using the below snippet instead? Will make for more readable and we will not have to perform additional look-ups.
```
pendingTaskCountPerTaskType.forEach(pendingTaskType, pendingTaskValue) -> {
    value.append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
    value.append(new MasterPendingStatus(pendingTaskType, pendingTaskValue).serialize());
  }
});
```

Reply:
Yes. This is more readable. Changed it.